apps/rush-lib/src/cli/utilities/test/ShrinkwrapFile.test.ts

```diff
@@ -29,25 +30,43 @@ describe('npm ShrinkwrapFile', () => {
   });
 });
 
-describe('pnpm ShrinkwrapFile', () => {
-  const filename: string = path.resolve(path.join(
-    __dirname, '../../../../src/cli/utilities/test/shrinkwrapFile/shrinkwrap.yaml'));
-  const shrinkwrapFile: BaseShrinkwrapFile = ShrinkwrapFileFactory.getShrinkwrapFile('pnpm', filename)!;
+describe('pnpm', () => {
+  describe('ShrinkwrapFile', () => {
+    const filename: string = path.resolve(path.join(
+      __dirname, '../../../../src/cli/utilities/test/shrinkwrapFile/shrinkwrap.yaml'));
+    const shrinkwrapFile: BaseShrinkwrapFile = ShrinkwrapFileFactory.getShrinkwrapFile('pnpm', filename)!;
 
-  it('verifies root-level dependency', () => {
-    assert.isTrue(shrinkwrapFile.hasCompatibleTopLevelDependency('jquery', '>=2.0.0 <3.0.0'));
-    assert.isFalse(shrinkwrapFile.hasCompatibleTopLevelDependency('q', '~1.5.0'));
-  });
+    it('verifies root-level dependency', () => {
+      assert.isTrue(shrinkwrapFile.hasCompatibleTopLevelDependency('jquery', '>=2.0.0 <3.0.0'));
+      assert.isFalse(shrinkwrapFile.hasCompatibleTopLevelDependency('q', '~1.5.0'));
+    });
 
-  it('verifies temp project dependencies', () => {
-    assert.isTrue(shrinkwrapFile.hasCompatibleDependency('jquery', '>=2.0.0 <3.0.0', '@rush-temp/project1'));
-    assert.isTrue(shrinkwrapFile.hasCompatibleDependency('q', '~1.5.0', '@rush-temp/project2'));
-    assert.isFalse(shrinkwrapFile.hasCompatibleDependency('left-pad', '~9.9.9', '@rush-temp/project1'));
-  });
+    it('verifies temp project dependencies', () => {
+      assert.isTrue(shrinkwrapFile.hasCompatibleDependency('jquery', '>=2.0.0 <3.0.0', '@rush-temp/project1'));
+      assert.isTrue(shrinkwrapFile.hasCompatibleDependency('q', '~1.5.0', '@rush-temp/project2'));
+      assert.isFalse(shrinkwrapFile.hasCompatibleDependency('left-pad', '~9.9.9', '@rush-temp/project1'));
+    });
 
-  it('extracts temp projects successfully', () => {
-    const tempProjectNames: ReadonlyArray<string> = shrinkwrapFile.getTempProjectNames();
+    it('extracts temp projects successfully', () => {
+      const tempProjectNames: ReadonlyArray<string> = shrinkwrapFile.getTempProjectNames();
 
-    assert.deepEqual(tempProjectNames, ['@rush-temp/project1', '@rush-temp/project2']);
+      assert.deepEqual(tempProjectNames, ['@rush-temp/project1', '@rush-temp/project2']);
+    });
+  });
+
+  describe('extractVersionFromPnpmVersionSpecifier', () => {
+    it('extracts a simple version with no slashes', () => {
```

Comment:
>it [](start = 4, length = 2)

Might be useful to test some degenerate cases such as:

""
"/"
"//"
"/@/"
etc